src/Elastic.OpenTelemetry/DependencyInjection/ServiceCollectionExtensions.cs

```diff
@@ -37,12 +36,15 @@ public static class ServiceCollectionExtensions
 	/// </returns>
 	public static IOpenTelemetryBuilder AddElasticOpenTelemetry(this IServiceCollection serviceCollection, ElasticOpenTelemetryOptions options)
 	{
-		if (serviceCollection.Any(d => d.ServiceType == typeof(IHostedService) && d.ImplementationType == typeof(ElasticOpenTelemetryService)))
+		var descriptor = serviceCollection.SingleOrDefault(s => s.ServiceType == typeof(ElasticOpenTelemetryBuilder));
+
+		if (descriptor?.ImplementationInstance is not null)
 		{
-			// TODO - Can we avoid this by storing the instance on the builder (internal access)
-			var sp = serviceCollection.BuildServiceProvider();
-			return sp.GetService<ElasticOpenTelemetryBuilder>()!; //already registered as singleton
+			var builder = (ElasticOpenTelemetryBuilder)descriptor.ImplementationInstance;
```

Comment:
Replace the if above it with an `as Elastic opentelemetry builder builder` instead of the hard cast?

Reply:
Good shout!